src/pages/resultsView/SurvivalStoreHelper.ts

```diff
@@ -0,0 +1,29 @@
+import {Patient, ClinicalData} from "shared/api/generated/CBioPortalAPI";
+import {PatientSurvival} from "../../shared/model/PatientSurvival";
+
+export function getPatientSurvivals(survivalClinicalDataGroupByUniquePatientKey: any, patients: Patient[], 
+    targetUniquePatientKeys: string[], statusAttributeId: string, monthsAttributeId: string, 
+    statusFilter: (s: string) => boolean): PatientSurvival[] {
+
+    let patientSurvivals: PatientSurvival[] = [];
+    if (targetUniquePatientKeys) {
+        targetUniquePatientKeys.forEach((uniquePatientKey: string) => {
```

Comment:
minor quibble but i like to use reduce in a case like this because it make it more clear that the the goal of the enclosed code is to produce the return value

Reply:
Isn't `reduce` used in order to reduce an array to a single value? In here I create a new array by transforming the valid elements of the original array. So it's more like a combination of `filter` and `map`, I can convert to them if you think it would be more readable.

Reply:
i think new array with potentially fewer elements qualifies semantically as a "reduction" of original.  you're right _filterMap would be better but I don't think it exists.  the key point to me is that it enapsulates the transformation and assigns the result to a value.  whereas, forEach, you have to actually read the forEach to understand that what you're doing is pushing values to a variable outside the foreach.

Reply:
I see, done.